### PR TITLE
Pulls out TrajectoryState into MapBuilderBridge.

### PR DESCRIPTION
**Pull request by damonkohler**

### CODE REVIEW

File: cartographer_ros/cartographer_ros/node.cc [hunk 1]
```diff
@@ -206,69 +206,57 @@ void Node::PublishSubmapList(const ::ros::WallTimerEvent& unused_timer_event) {
...
+      // published poses to advance.
+      stamped_transform.header.stamp = ros::Time::now();
+    }
+
+    if (trajectory_state.published_to_tracking != nullptr) {
+      if (options_.provide_odom_frame) {
+        std::vector<geometry_msgs::TransformStamped> stamped_transforms;
+
+        stamped_transform.header.frame_id = options_.map_frame;
+        stamped_transform.child_frame_id = options_.odom_frame;
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/node.cc]: Should we add a `TODO` that the `odom_frame` (and also `published_frame`?) must be per-trajectory when we fully support the multi-robot use case?

File: cartographer_ros/cartographer_ros/node.cc [hunk 2]
```diff
@@ -179,7 +179,7 @@ void Node::Initialize() {
       &Node::PublishSubmapList, this));
   wall_timers_.push_back(node_handle_.createWallTimer(
       ::ros::WallDuration(options_.pose_publish_period_sec),
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/node.cc]: Should `pose_publish_period_sec` be renamed, now that the function talks about 'trajectory states'?

File: cartographer_ros/cartographer_ros/map_builder_bridge.cc [hunk 3]
```diff
@@ -127,12 +127,33 @@ MapBuilderBridge::BuildOccupancyGrid() {
...
 }
 
-SensorBridge* MapBuilderBridge::sensor_bridge(const int trajectory_id) {
-  return sensor_bridges_.at(trajectory_id).get();
+std::unordered_map<int, MapBuilderBridge::TrajectoryState>
+MapBuilderBridge::GetTrajectoryStates() {
+  std::unordered_map<int, TrajectoryState> trajectory_states;
+  for (const auto& entry : sensor_bridges_) {
+    const int trajectory_id = entry.first;
+    const auto& sensor_bridge = entry.second;
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/map_builder_bridge.cc]: `auto` -> `SensorBridge`? `*entry.second`?

Comment (damonkohler) [inline cartographer_ros/cartographer_ros/map_builder_bridge.cc]: Done.

File: cartographer_ros/cartographer_ros/node.cc (hunk 1 above)
Comment (damonkohler) [inline cartographer_ros/cartographer_ros/node.cc]: Done.

File: cartographer_ros/cartographer_ros/node.cc (hunk 2 above)
Comment (damonkohler) [inline cartographer_ros/cartographer_ros/node.cc]: Let's leave this one alone for now until the API settles. We publish lots of things in addition to pose using this period.

**Pull request merged**